### PR TITLE
Doc improvements, fixes

### PR DESCRIPTION
**Pull request by cmarqu**
The fixed duplicated misrendering can be seen in e.g. 
https://external-builds.readthedocs.io/html/cocotb/1184/library_reference.html#cocotb.handle.RegionObject

### CODE REVIEW

File: cocotb/handle.py [hunk 1]
```diff
@@ -66,11 +70,6 @@ class SimHandleBase(object):
         }
 
     def __init__(self, handle, path):
-        """
-        Args:
-            handle (int): The GPI handle to the simulator object.
-            path (str): Path to this handle, ``None`` if root.
-        """
```
Comment (eric-wieser) [inline cocotb/handle.py]: Not a huge fan of moving docstrings away from the signature they describe, but maybe there's no way to make it render otherwise.

Comment (cmarqu) [reply cocotb/handle.py]: Maybe there is a way to use this docstring inheritance to reduce duplication, but for now I wanted to have the end result to not look bad.

Comment (eric-wieser) [reply cocotb/handle.py]: Yeah, the issue here is that the subclass repeats the args - removing it there would work too - I'll give that a try in a separate PR.

Comment (eric-wieser) [reply cocotb/handle.py]: Done in #1229.

Comment (cmarqu) [reply cocotb/handle.py]: Rebased to only include the rest of the fixes in this PR.

**Pull request merged**